core/input/input.cpp

```diff
@@ -1029,6 +1029,14 @@ void Input::parse_input_event(const Ref<InputEvent> &p_event) {
 	}
 }
 
+void Input::flush_frame_parsed_events() {
+	_THREAD_SAFE_METHOD_
+
+#ifdef DEBUG_ENABLED
+	frame_parsed_events.clear();
+#endif
+}
```

Comment:
Given that `frame_parsed_events` is a debug-only thing, I'd suggest making all these changes debug-only as well, so the function doesn't even exist in non-debug builds.

Aside, since "flush" is understood as "send events" in this context, I'd suggest a different name.

```suggestion
#ifdef DEBUG_ENABLED
void Input::clear_frame_parsed_events() {
	_THREAD_SAFE_METHOD_

	frame_parsed_events.clear();
}
#endif
```

Reply:
Makes sense to me to enclose all this (and `main.cpp` change) in `DEBUG_ENABLED`.